js/data/load_geometry.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+var EXTENT = require('./buffer').EXTENT;
+
+/**
+ * Loads a geometry from a VectorTileFeature and scales it to the common extent
+ * used internally.
+ * @private
+ */
+module.exports = function loadGeometry(feature) {
+    var scale = EXTENT / feature.extent;
+    var geometry = feature.loadGeometry();
+    for (var r = 0; r < geometry.length; r++) {
+        var ring = geometry[r];
+        for (var p = 0; p < ring.length; p++) {
+            var point = ring[p];
+            // round here because mapbox-gl-native uses integers to represent
+            // points and we need to do the same to avoid renering differences.
+            Math.round(point.x *= scale);
+            Math.round(point.y *= scale);
```

Comment:
That doesn't look right. `Math.round` is immutable, so `point` will remain unrounded. Should be:

```js
point.x = Math.round(point.x * scale);
point.x = Math.round(point.y * scale)
```

Also easier to read.